flang/include/flang/Lower/PFTBuilder.h

```diff
@@ -599,9 +599,11 @@ struct FunctionLikeUnit : public ProgramUnit {
   }
 
   /// Return a reference to the subprogram symbol of this FunctionLikeUnit.
+  /// This should not be called if the FunctionLikeUnit is the main program
+  /// since anonymous main programs do not have a symbol.
   const semantics::Symbol &getSubprogramSymbol() const {
     auto *symbol = entryPointList[activeEntry].first;
-    assert(symbol && "not inside a procedure");
+    assert(symbol && "not inside a procedure; do not call on main program.");
```

Comment:
We could even make this a hard error. It would help anyone who builds with assertions disabled.

Reply:
Updated to use `llvm::report_fatal_error`.